subsys/bluetooth/controller/ll_sw/ull_llcp.c

```diff
@@ -279,6 +279,38 @@ void llcp_tx_resume_data(struct ll_conn *conn, enum llcp_tx_q_pause_data_mask re
 	}
 }
 
+void llcp_rx_node_retain(struct proc_ctx *ctx)
+{
+	LL_ASSERT(ctx->node_ref.rx);
+
+	/* Mark RX node to NOT release */
+	ctx->node_ref.rx->hdr.type = NODE_RX_TYPE_RETAIN;
+
+	/* store link element reference to use once this node is moved up */
+	ctx->node_ref.rx->hdr.link = ctx->node_ref.link;
+}
+
+void llcp_nodes_release(struct ll_conn *conn, struct proc_ctx *ctx)
```

Comment:
is this functionality tested in unit or babblesim tests, for example tests/bsim/bluetooth/ll/conn/tests_scripts/basic_conn20_split.sh?

Reply:
It is tested in the basic_conn20_split - in the sense that I saw the test failing (due to inability to alloc RX nodes in the scanner) without the release flow. And I noticed - by inspecting printk's during dev that alloc'ed RX nodes were correctly released after applying the change